react/week3/my-app/src/index.js

```diff
@@ -0,0 +1,183 @@
+    return <li className={ (props.todoitem.completed) ? "done": "not-done" } onSubmit={props.Updatedesc}>  
+      <div> {details} </div>
+      <button onClick={props.EditTodo} >{editstate ? "Update" : "Edit"}</button>
+      <button onClick={props.DeleteTodo} >Delete</button>
+      <input type="checkbox" checked={props.todoitem.completed} onChange={props.ToggleCompleted}/>
+      <label></label> 
+  </li>;
+}
+
+function fetchlist(){
+  const URL='https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw';
+  return fetch(URL).then(response=>response.json())    
+}
+
+class TodoList extends React.Component {
+  state = {
+    index: '',
+    description: '',
+    deadline: '',
+    checked: false,
+    Todo: [],  
+  }
+
+  componentDidMount(){
+    this.fetchTodos();
+  }
+
+  fetchTodos = () => {
+    fetchlist().then(data => {
+      this.setState({ Todo: data})
+    })
+  }
+
+  AddTodoItems=(description, deadline)=>{    
+    if(!this.state.description){
+      alert("Please provide the description")
+    return;
+    }else if(!this.state.deadline){
+      alert("Please provide the date")
+    return;
+    }
+    this.setState({
+      Todo : [...this.state.Todo, {description : description, deadline : deadline, completed: false}],
+    })
+  };
+
+
+  ToggleCompleted = (currentIndex) => {
+    this.setState({
+      Todo : this.state.Todo.map((todo, index) => {
+        if (currentIndex === index){
+         return{
+           ...todo,
+         completed : !todo.completed, };
+        } else{
+          return todo;
+        }
+      }),
+    }); 
+  };
+
+  EditTodo = (props, description)=>{ 
+    this.setState({
+    Todo: this.state.Todo.map((list, index) => {
+      if (props === index) { 
+        return { ...list, editstate: !list.editstate, description: list.description,};
+      } else {
+        return list;
+        }
+      }),
+    });
+  }
+
+  Decisionmaker=(props)=>{
+    if(this.state.editstate === false){
+      return this.EditTodo;
+    }else{
+      return this.Updatedesc;
+    }
+  }
+
+  DeleteTodo = (currentIndex)=>{
+    this.setState({
+      Todo : this.state.Todo.filter((_todo, index)=> currentIndex !== index),
+    });
+  };
+  
+  UpdateTodo = (currentIndex, updates)=>{
+
+    (updates.description==='') ? (alert("Description can not be blank, please write something")) : 
+      this.setState({
+        Todo: this.state.Todo.map((list, index) => {
+          if (currentIndex === index) {
+            return { ...list, ...updates}       
+            } else {
+              return list;
+            }
+        }),
+      });
+  };
+ 
+  handleChange=(event)=>{   
+    this.setState({description: event.target.value});
+  }
+
+  handletimeChange=(event)=>{
+    let fulldate = moment(event).format("DD/MM/YYYY")
+    this.setState({deadline: fulldate});
+    let todaysdate = new Date();
+    if(event < todaysdate)
```

Comment:
One question for the second point. I should not use event to compare because of it's name? Because there is date inside event. Should i store in a variable for comparing
OR
i should not use event at all, with or without variable and use this: ==> moment(event)._d

i will use moment(event)._d for now but this question is for understanding why should i do that.

Reply:
I reviewed this part: you are right: semantically, this code is correct, I am sorry. My confusion came from the name `event`, that I understood to be a js "event" (the one with `.target.value`). However, this event is indeed just a date.

So, yes, my suggestion is to rename `event` on that function, to "selectedDate" or something that does not collide with the `event` traditionally used on js events. :)